src/main/java/nl/tudelft/contextproject/gui/PreviewController.java

```diff
@@ -40,133 +42,220 @@
     @FXML private Rectangle highlight4;
-    
-    private int currentShot;
+    @FXML private Label error;
+
+    private SimpleIntegerProperty currentShot;
+    private int currentSelected;
     private Timeline timeline;
     private ArrayList<Shot> shots;
     private ArrayList<ImageView> views;
 
     @FXML
     private void initialize() {
+        currentSelected = 1;
+
         actualCam.setImage(new Image("main/resources/placeholder_picture.jpg"));
         Image imgOne = new Image("main/resources/placeholder_picture.jpg");
         Image imgTwo = new Image("main/resources/test2.jpg");
         Image imgThree = new Image("main/resources/test3.jpg");
         Image imgFour = new Image("main/resources/test4.jpg");
-        Image imgFive = new Image("main/resources/test5.jpg");
-        Image imgSix = new Image("main/resources/test6.jpg");
-        
+
         viewOne.setImage(imgOne);
         viewTwo.setImage(imgTwo);
         viewThree.setImage(imgThree);
         viewFour.setImage(imgFour);
-        
-        views = new ArrayList();
+
+        views = new ArrayList<ImageView>();
         views.add(viewOne);
         views.add(viewTwo);
         views.add(viewThree);
         views.add(viewFour);
-        
+
         CameraSettings dummySettings = new CameraSettings();
         Camera dummyCamera = new Camera();
-        
+
         InstantPreset presetOne = new InstantPreset(dummySettings, 1);
         InstantPreset presetTwo = new InstantPreset(dummySettings, 2);
         InstantPreset presetThree = new InstantPreset(dummySettings, 3);
         InstantPreset presetFour = new InstantPreset(dummySettings, 4);
         InstantPreset presetFive = new InstantPreset(dummySettings, 5);
         InstantPreset presetSix = new InstantPreset(dummySettings, 6);
-        
+
         presetOne.setImageLocation("main/resources/placeholder_picture.jpg");
         presetTwo.setImageLocation("main/resources/test2.jpg");
         presetThree.setImageLocation("main/resources/test3.jpg");
         presetFour.setImageLocation("main/resources/test4.jpg");
         presetFive.setImageLocation("main/resources/test5.jpg");
         presetSix.setImageLocation("main/resources/test6.jpg");
-        
+
         Shot shotOne = new Shot(1, dummyCamera, presetOne);
         Shot shotTwo = new Shot(2, dummyCamera, presetTwo);
         Shot shotThree = new Shot(3, dummyCamera, presetThree);
         Shot shotFour = new Shot(4, dummyCamera, presetFour);
         Shot shotFive = new Shot(5, dummyCamera, presetFive);
         Shot shotSix = new Shot(6, dummyCamera, presetSix);
-        
-        shots = new ArrayList();
+
+        shots = new ArrayList<Shot>();
         shots.add(shotOne);
         shots.add(shotTwo);
         shots.add(shotThree);
         shots.add(shotFour);
         shots.add(shotFive);
         shots.add(shotSix);
+
+        currentShot = new SimpleIntegerProperty(1);
+        final ChangeListener changeListener = new ChangeListener() {
```

Comment:
You are using raw types here. You should parametrize this reference, since ChangeListener and ObservableValue are generic types: ChangeListener<T>.